### ISSUE TITLE
Scoop doesn't install correctly in Powershell 7.1

### ISSUE THREAD

**Issue by BaronKrause** [ScoopInstaller/Scoop#3952]
This might very well be a bug with the Powershell 7.1 preview, but figured I would report it anyway in case its an intended change on their end.

Installing scoop via Powershell 7.1 doesn't work correctly, it just "installs" without prompting for  "Set-ExecutionPolicy RemoteSigned -scope CurrentUser" (on new installs, assuming you haven't set it already). Even if you run "Set-ExecutionPolicy RemoteSigned -scope CurrentUser" manually in 7.1 it simply goes to the next line without ever giving you a y/n prompt, essentially doing nothing.

You can then install software, but after reboot nothing works, and Powershell just complains that there is nothing by that name for anything, including scoop. It doesn't make start menu shortcuts or set the path env variable either. Reinstalling fails too for the same cmdlet not found issue.

Running the Set-Execution command and scoop/software install in normal Powershell will work as it should however.

I tested and after running "Set-ExecutionPolicy RemoteSigned -scope CurrentUser" and then installing Scoop in normal Powershell, then switching back to 7.1 to install git, scoop still has some issues. The software installs, but it will not make start menu icons still.

**Comment by berwyn** [comment]
I played with this a bit and it seems like the `env` function is behaving oddly on PS7.1

I pulled the function into a local shell, using `env('test', 1)` to set the variable, and with a subsequent `env('test')` to retrieve it and got a null. If I set it with a full-on `[Environment]::SetEnvironmentVariable('test', 1, 'User')`, then `env('test')` returns the correct value. I've observed similar results with `'Machine'` level variables, where calling the C# API directly to set works, and using the PowerShell function to get also works.

tl;dr -- it doesn't seem like `env` can correctly set variables in PS7.1, however it can retrieve them. I'll see if I can dig in further to this.

**Comment by dim5** [comment]
7.1 has been released, and this issue still seems to be present

**Comment by cprivite** [comment]
Can confirm here too, does not install correctly with 7.1

**Comment by strange-life** [comment]
On my computer, the problem only occurs when I install from the Microsoft Store

**Comment by KDrakoulaki** [comment]
Just tried with 7.2 preview and still the same - has anyone found a workaround?

**Comment by megakid** [comment]
I'm also getting this from Powershell (now v7.1.1) installed from Microsoft Store (the recommended way by Microsoft)

**Comment by lsgroi** [comment]
As workaround, I have used Windows PowerShell rather than PowerShell Core

**Comment by IAmIlliest** [comment]
Same problem PS Core 7.1.3

**Comment by altavir** [comment]
The same problem here. Environment variables are not set in PS Core 7. It works fine on regular powershell.

**Comment by leni8ec** [comment]
Same problem.

PowerShell 7.1.3 - has been **installed from MicrosoftStore**.
In preinstalled Powershell (blue) - all is ok!

`Set-ExecutionPolicy RemoteSigned -scope CurrentUser` - does not help.

Start menu shortcuts - has been re/created in **PowerShell cache dir** (after `scoop reset *`):
```
C:\Users\[username]\AppData\Local\Packages\Microsoft.PowerShell_8wekyb3d8bbwe\LocalCache\Roaming\Microsoft\Windows\Start Menu\Programs\Scoop Apps\
```

Instead of:
```
C:\Users\[username]\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Scoop Apps\
```

**Comment by leni8ec** [comment]
It looks like it has to do with [limitations](https://docs.microsoft.com/en-gb/powershell/scripting/install/installing-powershell-core-on-windows?view=powershell-7.1#installing-from-the-microsoft-store) of the **MSIX** packages (**Microsoft Store** apps)

i.e. PowerShell in this case - is launched as if in a **sandbox** and cannot interact with user variables and data.

**Solution:** Install PowerShell via [winget](https://docs.microsoft.com/en-gb/powershell/scripting/install/installing-powershell-core-on-windows?view=powershell-7.1#install-powershell-via-the-windows-package-manager) or [manual](https://docs.microsoft.com/en-gb/powershell/scripting/install/installing-powershell-core-on-windows?view=powershell-7.1#download-the-installer-package).

**Comment by kele14x** [comment]
I got same problem in my machine. 

Seems that if you run `Set-ExecutionPolicy RemoteSigned -scope CurrentUser` in **Windows PowerShell** instead of **PowerShell 7.1**. Scoop will work normally. It will set path, create folder under **C:\Users\\[username]** and shortcut in program menu.

But if you does not set executionpolicy in Windows PowerShell (even you do in PowerShell 7.1), scoop instead create folder under **C:\Users\\[username]**, but does not set path, and shortcut goes to **C:\Users\\[username]\\AppData\Local\Packages**. Which will lose after reboot.

Set executionpolicy in WIndows PowerShell and switch back to PowerShell core should still work.

**Comment by filipponova** [comment]
I following the steps suggested @kele14x and the scoop back working.  Tks @kele14x :+1:

**Comment by rashil2000** [comment]
https://github.com/ScoopInstaller/Scoop/issues/4282
